2-mandatory/3-groceries.js

```diff
@@ -27,13 +27,21 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
-
+for (let key in weeklyMealPlan) {
+  weeklyMealPlan[key].forEach(items => {
+    if (!weeklyGroceriesToBuy.includes(items)) {
+      weeklyGroceriesToBuy.push(items)
+    }
+  }
+    )
+}
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
+Object.keys(weeklyMealPlan).forEach((key, index) => index >= 5 ? weekendGroceriesToBuy.push(...weeklyMealPlan[key]) : undefined);
```

Comment:
I would not assume that Saturday and Sunday will have an index > 5. Better to look at the key name

It looks like you will be pusing undefined into the array. Rather than suing Object. keys, you could use a for.. in loop